src/disassembler.rs

```diff
@@ -0,0 +1,17 @@
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use std::fmt;
+
+// Supported architectures
+enum Architecture {
+    X86,
+    X86_64,
```

Comment:
Yes, but I think we could still differentiate between the two using the mode.
Probably other disassemblers make a distinction between the two, and I think it makes sense to differentiate them.